tests/FSharp.Compiler.ComponentTests/Signatures/TestCasesForGenerationRoundTrip/array.fsx

```diff
@@ -0,0 +1,1151 @@
+    do loop4 0 0
+    do loop4 0 -1
+    do loop4 10  9
+    do for start in -3 .. 3 do for finish in -3 .. 3 do loop4 start finish
+
+    let loop5 a N = 
+       let mutable x = 0 in
+       // In this loop, the types of 'a' and 'N' are not known prior to the loop
+       for i in (min a a) ..  2 .. (min N N) do
+          x <- x + 1
+       done;
+       check (sprintf "ffclkrerev92-%A" (a,N))  x ((if N < a then 0 else N - a + 2) / 2)
+
+    do loop5 0 10
+    do loop5 0 0
+    do loop5 0 -1
+    do loop5 10  9
+    do for start in -3 .. 3 do for finish in -3 .. 3 do loop5 start finish
+
+
+    let loop6 a N = 
+       let mutable x = 0 in
+       // In this loop, the types of 'a' and 'N' are not known prior to the loop
+       for i in (min a a) ..  200 .. (min N N) do
+          x <- x + 1
+       done;
+       check (sprintf "ffclkrerev93-%A" (a,N)) x ((if N < a then 0 else N - a + 200) / 200)
+
+    do loop6 0 10
+    do loop6 0 0
+    do loop6 0 -1
+    do loop6 10  9
+    do for start in -3 .. 3 do for finish in -3 .. 3 do loop6 start finish
+
+
+    let loop7 a step N = 
+       let mutable x = 0 in
+       // In this loop, the types of 'a' and 'N' are not known prior to the loop
+       for i in (min a a) ..  step .. (min N N) do
+          x <- x + 1
+       done;
+       check (sprintf "ffclkrerev95-%A" (a,step,N)) x (if step < 0 then (if a < N then 0 else (a - N + abs step) / abs step) else (if N < a then 0 else N - a + step) / step)
+
+    do loop7 0 1 10
+    do loop7 0 -1 0
+    do loop7 0 2 -1
+    do loop7 10  -2 9
+    do for start in -3 .. 3 do for finish in -3 .. 3 do for step in [-2; -1; 1; 2] do loop7 start step finish
+
+    let loop8 a N = 
+       let mutable x = 0 in
+       // In this loop, the types of 'a' and 'N' are not known prior to the loop
+       for i in (min a a) ..  -1 .. (min N N) do
+          x <- x + 1
+       done;
+       check (sprintf "ffclkrerev96-%A" (a,N))  x (abs (if a < N then 0 else (a - N + 1) / 1))
+
+    do loop8 0 10
+    do loop8 0 0
+    do loop8 0 -1
+    do loop8 10 9
+    do for start in -3 .. 3 do for finish in -3 .. 3 do loop8 start finish
+
+module bug872632 =
+    type MarkerStyle = 
+        | None      = 0
+        | Square    = 1
+        | Circle    = 2
+        | Diamond    = 3
+        | Triangle    = 4
+        | Triangle1   = 10
+        | Cross    = 5
+        | Star4    = 6
+        | Star5    = 7
+        | Star6    = 8
+        | Star10    = 9
+
+     
+
+    module Foo =    
+        let x = [| 
+                    MarkerStyle.Circle
+                    MarkerStyle.Cross
+                    MarkerStyle.Star6
+                    MarkerStyle.Diamond
+                    MarkerStyle.Square        
+                    MarkerStyle.Star10
+                    MarkerStyle.Triangle
+                    MarkerStyle.Triangle1
+                |] 
+
+    do check "bug872632" Foo.x.Length 8
+
+module CheckUnionTypesAreSealed =
+    open System
+#if NETCOREAPP
```

Comment:
Still required here?

Reply:
The code has been in general kept unless things were failing.
This is a harmless polyfill , keeping it.